js/client-chat.js

```diff
@@ -958,6 +958,13 @@
 					});
 					break;
 
+				case 'remove':
+					if (app.user.get('isStaff')) break;
```

Comment:
Default should probably hide the messages but add a small-font notice "Some spam messages were hidden, click here to restore."